pkg/webhookconfig/configmanager.go

```diff
@@ -703,21 +704,25 @@ func (m *webhookConfigManager) compareAndUpdateWebhook(webhookKind, webhookName
 }
 
 func (m *webhookConfigManager) updateStatus(namespace, name string, ready bool) error {
-	update := func(meta *metav1.ObjectMeta, spec *kyverno.Spec, status *kyverno.PolicyStatus) bool {
+	update := func(meta *metav1.ObjectMeta, p kyverno.PolicyInterface, status *kyverno.PolicyStatus) bool {
 		copy := status.DeepCopy()
-		requested, _, activated := autogen.GetControllers(meta, spec, m.log)
+		requested, _, activated := autogen.GetControllers(meta, p.GetSpec(), m.log)
 		status.SetReady(ready)
 		status.Autogen.Requested = requested
 		status.Autogen.Activated = activated
-		status.Rules = spec.Rules
+		if toggle.AutogenInternals() {
+			status.Rules = autogen.ComputeRules(p)
+		} else {
+			status.Rules = nil
+		}
```

Comment:
@JimBugwadia with this change, rules are written in the status only when autogen internals is enabled.

Reply:
@eddycharly - thanks! That is good, but as I asked in the slack discussion why do we need to write the entire rule to the status even in this case? 

I can some value in troubleshooting as we are developing the feature, but feel it will be confusing for users. I feel it makes the YAML harder read (especially as the metadata and other junk added), and also makes it harder to see when a problem occurs with autogen. 

I suggest we add a simpler summary for autogen to the status, with relevant errors, and move displaying the rule to a L4 log so its available when / if we need it.

Reply:
I also agree. As a user of this functionality. I really am interested in seeing just two things: 1) that auto-gen is enabled for the controllers I expect; and 2) if something isn't working as advertised, I am told about it without enough info to know what to do.

Reply:
TBH I’m not convinced. Logs are not always accessible.
Having the full set of rules for a given policy easily accessible looks useful to me.

I agree that we can improve the status and indicate what failed more clearly though.